vcd/resource_vcd_edgegateway.go

```diff
@@ -196,44 +314,45 @@ func setEdgeGatewayValues(d *schema.ResourceData, egw govcd.EdgeGateway) error {
 	//if err != nil {
 	//	return err
 	//}
+
 	return nil
 }
 
-// Fetches information about an existing edge gateway for a data definition
-func resourceVcdEdgeGatewayRead(d *schema.ResourceData, meta interface{}) error {
-	log.Printf("[TRACE] edge gateway read initiated")
-
-	vcdClient := meta.(*VCDClient)
-
-	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "name")
+// setLoadBalancerData is a convenience function to handle load balancer settings on edge gateway
+func setLoadBalancerData(d *schema.ResourceData, egw govcd.EdgeGateway) error {
+	lb, err := egw.GetLBGeneralParams()
 	if err != nil {
-		d.SetId("")
-		return nil
+		return fmt.Errorf("unable to read general load balancer settings: %s", err)
 	}
 
-	err = setEdgeGatewayValues(d, edgeGateway)
+	d.Set("lb_enabled", lb.Enabled)
+	d.Set("lb_acceleration_enabled", lb.AccelerationEnabled)
+	d.Set("lb_logging_enabled", lb.Logging.Enable)
+	d.Set("lb_loglevel", lb.Logging.LogLevel)
 
-	log.Printf("[TRACE] edge gateway read completed: %#v", edgeGateway.EdgeGateway)
-	return err
+	return nil
 }
 
-// Deletes a edge gateway, optionally removing all objects in it as well
-func resourceVcdEdgeGatewayDelete(d *schema.ResourceData, meta interface{}) error {
-	log.Printf("[TRACE] edge gateway delete started")
-
-	vcdClient := meta.(*VCDClient)
-
-	vcdClient.lockEdgeGateway(d)
-	defer vcdClient.unlockEdgeGateway(d)
-
-	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "name")
+// updateLoadBalancer updates general load balancer configuration
+func updateLoadBalancer(d *schema.ResourceData, egw govcd.EdgeGateway) error {
+	lbEnabled := d.Get("lb_enabled").(bool)
+	lbAccelerationEnabled := d.Get("lb_acceleration_enabled").(bool)
+	lbLoggingEnabled := d.Get("lb_logging_enabled").(bool)
+	lbLogLevel := d.Get("lb_loglevel").(string)
+	_, err := egw.UpdateLBGeneralParams(lbEnabled, lbAccelerationEnabled, lbLoggingEnabled, lbLogLevel)
 	if err != nil {
-		d.SetId("")
-		return fmt.Errorf("error fetching edge gateway details %#v", err)
+		return fmt.Errorf("unable to update general load balancer settings: %s", err)
 	}
 
-	err = edgeGateway.Delete(true, true)
+	return nil
+}
 
-	log.Printf("[TRACE] edge gateway deletion completed\n")
-	return err
+// isEdgeGatewayLbConfigured checks if any of load balancer related settings are set
```

Comment:
I suggest adding that this only applies to LB **general parameters**.

Reply:
Getting rid of it as we are going to set default load balancer settings always. This check doesn't help anymore when defaults are used.